modules/placement-driver/src/integrationTest/java/org/apache/ignite/internal/placementdriver/ItPrimaryReplicaChoiceTest.java

```diff
@@ -159,15 +210,140 @@ public void testLockReleaseWhenPrimaryChange() throws Exception {
 
         IgniteImpl ignite = node(primary);
 
-        InternalTransaction rwTx = (InternalTransaction) ignite.transactions().begin();
+        int hashIdxId = getIndexId(HASH_IDX);
+        int sortedIdxId = getIndexId(SORTED_IDX);
+
+        InternalTransaction rwTx = (InternalTransaction) node(0).transactions().begin();
+        InternalTransaction roTx = (InternalTransaction) node(0).transactions().begin(new TransactionOptions().readOnly(true));
 
         assertTrue(tbl.recordView().insert(rwTx, Tuple.create().set("key", 42).set("val", "val 42")));
 
+        BinaryTuple idxKey = new BinaryTuple(1, new BinaryTupleBuilder(1).appendString("preload val").build());
+
+        // Start cursors in the RW transaction.
+        scanSingleEntryAndLeaseCursorOpen(tbl, rwTx, null, null);
```

Comment:
What do you mean by "lease" in scanSingleEntryAndLeaseCursorOpen? Probably it's "leave".

Reply:
I mean, the method leaves the cursor open to check that the cursor will clear.